File: leetcode/1161.最大层内元素和.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func maxLevelSum(root *TreeNode) int {
    mx := math.MinInt
    idx := -1
    step := 1
    q := []*TreeNode{root}
    for len(q) > 0 {
        tot := 0
        m := len(q)
        for i := 0; i < m; i++ {
            t := q[0]
            tot += t.Val
            q = q[1:]
            if t.Left != nil {
                q = append(q, t.Left)
            }
            if t.Right != nil {
                q = append(q, t.Right)
            }
        }
        if tot > mx {
            mx = tot
            idx = step
        }
        step += 1
    }
    return idx
}

File: leetcode/1161.最大层内元素和.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def maxLevelSum(self, root: Optional[TreeNode]) -> int:
        que=deque()
        que.appendleft(root)
        res=-1e9
        fl=0
        dep=0
        while que:
            n=len(que)
            ans=0
            dep+=1
            for i in range(n):
                a=que.pop()
                ans+=a.val
                if a.left:
                    que.appendleft(a.left)
                if a.right:
                    que.appendleft(a.right)
            if ans>res:
                res=ans
                fl=dep
        return fl